platform_versioned_docs/version-24.1/data_studios/custom-envs.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: "Custom environments"
+description: "Custom environments for Data Studios"
+date: "1 Oct 2024"
+tags: [data, studios]
+---
+
+In addition to the Seqera-provided container template images, you can provide your own custom container environments, either by augmenting the Seqera-provided images with a list of Conda packages that you provide or by providing your own base container template image.
+
+## Conda packages
+
+### Prerequisites
+
+- Wave must be configured. For more information, see [Wave containers][wave].
+
+### Conda package syntax {#conda-package-syntax}
+
+When you add a new data studio, you can customize its configuration to install a list of Conda packages to the template. The supported schema is identical to that used by the Conda `environment.yml` environment file. For more information on the Conda environment file, see [Creating an environment file manually][env-manually].
+
+```yaml title="Example environment.yml file"
+name: example-env
+channels:
+  - defaults
+  - conda-forge
+dependencies:
+  - python=3.7
+  - numpy
+  - pandas
+  - scikit-learn
+  - pip:
+    - matplotlib
+    - seaborn
+    - jupyter
+```
+
+To create a data studio with custom Conda packages, see [Add a data studio][add-ds].
+
+## Custom container template image {#custom-containers}
+
+For advanced use cases, you can build your own container template image.
+
+:::note
+Amazon Elastic Container Registry (ECR) is the only supported private container registry.
+:::
+
+### Prerequisites
+
+- Access to a container image repository, either a public container registry or in a private Amazon ECR repository
+- A container template image
+
+### Dockerfile configuration {#dockerfile}
+
+For your custom template container image, you must use a Seqera provided base image and include several additional build steps for compatibility with Data Studios.
+
+#### Ports
+
+The container must use the value of the `CONNECT_TOOL_PORT` environment variable as the listening port for whatever interactive software that you include in your custom container.
+
+#### Signals
+
+Upon termination, the container's main process must handle the `SIGTERM` signal and perform any necessarily cleanup. After a 30 second grace period, the container receives the `SIGKILL` signal.
+
+#### Minimal Dockerfile
+
+The minimal Dockerfile includes directives to accomplish the following:
+
+- Pulls a Seqera-provided base image with prerequisite binaries
+- Installs prerequisite packages, such as wget and fuse
```

Comment:
```suggestion
- Installs prerequisite packages, such as `wget` and `fuse`
```

Reply:
@robnewman, this is always a bit confounding, as for example the package name that is installed on say Debian is actually called _wget_, and the binary itself is called `wget`. Looking at the Dockerfile, we install these packages, which then provide the binaries of the same name.

This is similar to in K8s, where we might have a _pod_ that is configured with the Pod resource. Is it Pod or pod? The distinction isn't always clear. In this case, I think just _wget_ or _fuse_ is closer to the mark.

Reply:
Okay - sounds good. Ignore my suggestion.